### DISCUSSION TITLE
Custom FileUpload ,how to store the image in db

### DISCUSSION THREAD

**Discussion by KumarSubham101201** [filamentphp/filament#9521] [Help]
<x-dynamic-component
    :component="$getFieldWrapperView()"
    :field="$field"
>
    <div x-data="{ state: $wire.$entangle('{{ $getStatePath() }}') }">
    <x-filament::input.wrapper>
    <x-filament::input
        type="file"
        wire:model="name"
    />
</x-filament::input.wrapper>
    </div>
</x-dynamic-component>

i just create this custom component and use this in a resource file but it can't store the data in db
it store null

**Comment by sinnbeck** [comment]
I would highly recommend storing images on the disk. Databases arent made for storing images

**Reply by KumarSubham101201** [reply]
but i need the image path to store there, how can i show in the imagcolumn

**Comment by leandrocfe** [comment]
Why you are using a custom fileupload? Why not [file-upload](https://filamentphp.com/docs/3.x/forms/fields/file-upload)?

**Reply by KumarSubham101201** [reply]
bro coz i have a figma to match the design so there they use a custom file upload so i try to make a custom fileupload , bro i just create a custom component and extend from fileupload but i dont understand how to store the image url in the db ,can you give me some piece of code for reference.i know the default fileupload is better but i have some requirement and i am totally new to this
